qcodes/dataset/data_set.py

```diff
@@ -1167,6 +1176,194 @@ def __repr__(self) -> str:
 
         return "\n".join(out)
 
+    def _enqueue_results(
```

Comment:
Perhaps we should make this a public api?

Reply:
that's probably fair, espcially if it's guarded by checking what state the dataset is in - enqueue should only work in the "running" state similar to add_results.

On the other hand, the interfaces of "add_results" and "enqueue results" differ significantly - add_results does not know about qcodes parameters and the fact that things need to unpacked, which enqueue_results knows. So, i would refrain from making it public just for now.

Reply:
on the other hand, there's also a publci `add_result_to_queue` :) 

UPDATE: had a look, and that method is NOT used by any code :) moreover, it's behavior is totally included in the behavior of `add_result`. So, either in this or another PR, we need to remove this `add_result_to_queue` method.

Reply:
my conclusion - let's keep `_enqueue_results` private for now, and regardless of that guard it with checks for the "running" state of the DataSet.